Products/PleiadesEntity/browser/attestations.py

```diff
@@ -176,35 +177,60 @@ def rows(self, locations):
         wftool = self.wftool
         checkPermission = getSecurityManager().checkPermission
         credit_utils = self.context.unrestrictedTraverse('@@credit_utils')
-        for score, ob, nrefs in sorted(locations, reverse=False):
-            review_state = wftool.getInfoFor(ob, 'review_state')
-            item = ob.Title().decode('utf-8')
-            if 'copy' in ob.getId():
-                item += u" (copy)"
-            if checkPermission('View', ob):
-                link = u'<a class="state-%s" href="%s">%s</a>' % (
-                    review_state, ob.absolute_url(), item)
-            else:
-                link = u'<span class="state-%s">%s</span>' % (
-                    review_state, item)
-            if review_state != 'published':
-                user = credit_utils.user_in_byline(ob.Creator())
-                status = u' [%s by %s]' % (review_state, user['fullname'].decode('utf-8'))
-            else:
-                status = u''
-            innerHTML = [
-                u'<li id="%s_%s" class="placeChildItem Location" title="%s">' % (
-                    ob.getId(),
-                    where_tag,
-                    self.snippet(ob) + "; " + ob.Description().decode("utf-8"),
-                ),
-                self.prefix(ob),
-                link,
-                self.postfix(ob),
-                status,
-                u'</li>',
-            ]
-            output.append(u"\n".join(innerHTML))
+        location_types = [
+            "associated_modern",
+            "relocated_modern",
+            "central_point",
+            "legacy",
+            "representative",
+        ]
+        labeled_locations = defaultdict(set)
+        for _, ob, _ in sorted(locations, reverse=False):
+            current_location_types = ob.getLocationType()
+            found = False
+            for location_type in location_types:
```

Comment:
Maybe we should use set intersection here instead of the implicit nested loop created by `in`? The `n`s are small enough here that it probably doesn't much matter.

Reply:
The current code will take into account the ordering of `location_types`: the location will be under the first type found in that array. So if a location has both `associated_modern` and `relocated_modern` in its types it will be put among the former set. I'm not sure whether this is important or not, but that's how the pseudo code was written, and I followed that logic.

Reply:
This seems to be working fine as coded, so let's leave it as is.